src/bin/main.rs

```diff
@@ -342,13 +346,12 @@ fn main() {
         Ok(summary) => {
             if summary.has_operational_errors() {
                 1
+            } else if summary.has_diff && summary.write_mode == WriteMode::Check {
+                1
             } else if summary.has_parsing_errors() {
-                2
+                1
             } else if summary.has_formatting_errors() {
-                3
-            } else if summary.has_diff {
-                // should only happen in diff mode
-                4
+                1
```

Comment:
You could use `||` rather than having multiple `else if` branches here.

Reply:
Good point, will update later today.